assets/src/edit-story/output/utils/getFontDeclarations.js

```diff
@@ -60,30 +60,34 @@ const getFontDeclarations = (pages) => {
 
       const fontObj = serviceMap.get(family) || { family, variants: [] };
 
+      const contentVariants = getFontVariants(content);
+
       if (variants.length > 0) {
-        // A variant ("axis tuple" in Google Fonts terms) is a combination
-        // of font style and weight for a given font.
-        // The first item is a flag for italic.
-        // The second item is the numeric font weight.
-        // Example: [1, 700] for italic + bold
-        const variant = [Number(fontStyle === 'italic'), fontWeight || 400];
-
-        // Use closest variant as fallback and let browser do the math if needed.
-        // Examples:
-        // - If only [ [ 0, 200 ], [ 0, 400 ] ] exist, and
-        //   [ 1, 200] was requested, fall back to [ 0, 200 ].
-        // - If only [ [ 0, 400 ], [ 0, 800 ] ] exist, and
-        //   [ 1, 800] was requested, fall back to [ 0, 800 ].
-        // - If only [ [ 1, 400 ] ] exist, and
-        //   [ 0, 400] was requested, fall back to [ 1, 400 ].
-
-        const newVariant = getNearestTuple(variants, variant);
-        const fontObjHasVariant = hasTuple(fontObj.variants, newVariant);
-        const isValidVariant = hasTuple(variants, newVariant);
-
-        // Keeps list unique.
-        if (!fontObjHasVariant && isValidVariant) {
-          fontObj.variants.push(newVariant);
+        for (const [italic, weight] of contentVariants) {
+          // A variant ("axis tuple" in Google Fonts terms) is a combination
+          // of font style and weight for a given font.
+          // The first item is a flag for italic.
+          // The second item is the numeric font weight.
+          // Example: [1, 700] for italic + bold
+          const variant = [Number(italic), weight || 400];
```

Comment:
No need for this as `contentVariants` already contains the variants in the right format, no?

So you could do `for (const variant of contentVariants) {`

Reply:
Yes, true. Left it in for ensuring the `Number` + the `weight || 400` and the informative comment you left there! Can move the things around to use the `variant` directly though.

Reply:
We can move the comment above the `for` 👍 
I think it's fine to expect the right format from `contentVariants`